packages/opentelemetry-api/src/api/propagation.ts

```diff
@@ -0,0 +1,71 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Context } from '@opentelemetry/scope-base';
+import { Carrier } from '../context/propagation/carrier';
+import { HttpTextFormat } from '../context/propagation/HttpTextFormat';
+import { NOOP_HTTP_TEXT_FORMAT } from '../context/propagation/NoopHttpTextFormat';
+import { ContextAPI } from './context';
+
+const contextApi = ContextAPI.getInstance();
+
+/**
+ * Singleton object which represents the entry point to the OpenTelemetry Propagation API
+ */
+export class PropagationAPI {
+  private static _instance?: PropagationAPI;
+  private _propagator: HttpTextFormat = NOOP_HTTP_TEXT_FORMAT;
+
+  /** Empty private constructor prevents end users from constructing a new instance of the API */
+  private constructor() {}
+
+  /** Get the singleton instance of the Propagator API */
+  public static getInstance(): PropagationAPI {
+    if (!this._instance) {
+      this._instance = new PropagationAPI();
+    }
+
+    return this._instance;
+  }
+
+  /**
+   * Set the current propagator. Returns the initialized propagator
+   */
+  public initGlobalPropagator(propagator: HttpTextFormat): HttpTextFormat {
```

Comment:
nit: consistency with using `public` keyword

Reply:
to be clear, we want to always use or never use?